File: 코딩 기초 트레이닝/수 조작하기 2-p181925/수 조작하기 2.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181925

def solution(numLog):
    answer = ''
    prev = numLog[0]

    for next in range(len(numLog)-1):
        next = next+1

        diff = abs(numLog[next] - prev)
        if numLog[next] > prev:
            if diff == 1:
                answer += 'w'
            else:
                answer += 'd'
        else:
            if diff == 1:
                answer += 's'
            else:
                answer += 'a'

        prev = numLog[next]

    return answer

File: 코딩 기초 트레이닝/수 조작하기 2-p181925/수 조작하기 2.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181925

function solution(numLog) {
    const answer = [];
    let prev = numLog[0];

    for(let i = 1; i < numLog.length ; i++) {
        let next = numLog[i];
        const diff = Math.abs(next - prev);

        if(next > prev) {
            if(diff === 1) {
                answer.push('w');
            } else {
                answer.push('d');
            }
        } else {
            if(diff === 1) answer.push('s');
            else answer.push('a');
        }
        prev = next;
    }

    return answer.join('');
}
